src/Components/Aspire.Azure.AI.Inference/AspireAzureAIInferenceExtensions.cs

```diff
@@ -0,0 +1,178 @@
+                    if (!string.IsNullOrEmpty(settings.Key))
+                    {
+                        var credential = new AzureKeyCredential(settings.Key);
+                        return new ChatCompletionsClient(settings.Endpoint, credential, options);
+                    }
+                    else
+                    {
+                        return new ChatCompletionsClient(settings.Endpoint, settings.TokenCredential ?? new DefaultAzureCredential(), options);
+                    }
+                }
+            });
+        }
+
+        protected override void BindClientOptionsToConfiguration(IAzureClientBuilder<ChatCompletionsClient, AzureAIInferenceClientOptions> clientBuilder, IConfiguration configuration)
+        {
+#pragma warning disable IDE0200 // Remove unnecessary lambda expression - needed so the ConfigBinder Source Generator works
+            clientBuilder.ConfigureOptions(options => configuration.Bind(options));
+#pragma warning restore IDE0200
+        }
+
+        protected override void BindSettingsToConfiguration(ChatCompletionsClientSettings settings, IConfiguration configuration)
+            => configuration.Bind(settings);
+
+        protected override IHealthCheck CreateHealthCheck(ChatCompletionsClient client, ChatCompletionsClientSettings settings)
+            => throw new NotImplementedException();
+
+        protected override bool GetHealthCheckEnabled(ChatCompletionsClientSettings settings)
+            => false;
+
+        protected override bool GetMetricsEnabled(ChatCompletionsClientSettings settings)
+            => !settings.DisableMetrics;
+
+        protected override TokenCredential? GetTokenCredential(ChatCompletionsClientSettings settings)
+            => settings.TokenCredential;
+
+        protected override bool GetTracingEnabled(ChatCompletionsClientSettings settings)
+            => !settings.DisableTracing;
+    }
+
+    /// <summary>
+    /// Creates a <see cref="IChatClient"/> from the <see cref="ChatCompletionsClient"/> registered in the service collection.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <returns></returns>
+    public static ChatClientBuilder AddChatClient(this AspireChatCompletionsClientBuilder builder) =>
+        builder.Builder.Services.AddChatClient(services =>
+        {
+            var chatCompletionsClient = !string.IsNullOrEmpty(builder.ServiceKey) ?
+                services.GetRequiredService<ChatCompletionsClient>() :
+                services.GetRequiredKeyedService<ChatCompletionsClient>(builder.ServiceKey);
+
+            var result = chatCompletionsClient.AsIChatClient();
+
+            if (builder.DisableTracing)
+            {
+                return result;
+            }
+
+            var loggerFactory = services.GetService<ILoggerFactory>();
+            return new OpenTelemetryChatClient(result, loggerFactory?.CreateLogger(typeof(OpenTelemetryChatClient)));
```

Comment:
```suggestion
            return new ChatClientBuilder(result)
                .UseOpenTelemetry()
                .Build();
```

Is this the better approach then @davidfowl?

Reply:
I think so unless I'm missing something 😄